pkg/apis/v1beta1/ec2nodeclass_hash_test.go

```diff
@@ -0,0 +1,147 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1_test
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/imdario/mergo"
+	"github.com/samber/lo"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+	"github.com/aws/karpenter/pkg/test"
+)
+
+var _ = Describe("Hash", func() {
+	const staticHash = "16608948681250225098"
```

Comment:
This gives us a lot of value. This ensures that we still resolve to the same hash value so that if we make a change that in-advertantly changes the algorithm, we will catch it. If we don't do this, then we can make a change that updates the hash for everything and all of a sudden the next version causes the whole cluster to roll for everyone.

Reply:
Ah that's a great point